src/overlays/actors/ovl_player_actor/z_player.c

```diff
@@ -2307,114 +2333,131 @@ s32 func_80833CDC(PlayState* play, s32 index) {
     }
 }
 
-void func_80833DF8(Player* this, PlayState* play) {
+/**
+ * Handles the high level item usage and changing process.
+ *
+ * Tasks include:
+ *    - Put away a mask if it is not present on any C button
+ *    - Put away an item if it is not present on the B button or any C button
+ *    - Use an item on the B button or any C button if the corresponding button is pressed
+ *    - Keep track of the curent item button being held down
+ *
+ * Note: The item put-away/take-out process is not handled here.
+ */
+void Player_ItemHandler(Player* this, PlayState* play) {
     s32 maskItemAction;
     s32 item;
     s32 i;
 
     if (this->currentMask != PLAYER_MASK_NONE) {
         maskItemAction = this->currentMask - 1 + PLAYER_IA_MASK_KEATON;
-        if (!func_80833C98(C_BTN_ITEM(0), maskItemAction) && !func_80833C98(C_BTN_ITEM(1), maskItemAction) &&
-            !func_80833C98(C_BTN_ITEM(2), maskItemAction)) {
+
+        if (!Player_ItemIsItemAction(C_BTN_ITEM(0), maskItemAction) &&
+            !Player_ItemIsItemAction(C_BTN_ITEM(1), maskItemAction) &&
+            !Player_ItemIsItemAction(C_BTN_ITEM(2), maskItemAction)) {
```

Comment:
So having a mask on B (without having a mask on a C button) would not allow you to use it? Interesting

Reply:
I had no idea, but yes, just tested this and its true
(ofc, masks on B is only achievable through glitches)

Reply:
dont know whats happening here. I tested in 1.0 as child (since masks as adult crashes emu). it shouldnt be a version difference cause decomp is based on mq which is much after 1.2

Reply:
I believe in that video he is pressing mask and then talking to gossip stone before the game has a chance to take the mask off. you have a frame or two before it does so.

Reply:
ahh I see, so this is why it's removing the mask instantly, I thought it was preventing the mask from being used at all, like no frames of equipped then removing it (if that makes sense)

I tried on 1.2 and it's working like it should be 👍